homeassistant/components/telekomunikace_2n/config_flow.py

```diff
@@ -0,0 +1,142 @@
+"""Config flow for 2N Telekomunikace integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+from py2n import Py2NConnectionData, Py2NDevice
+from py2n.exceptions import (
+    DeviceApiError,
+    DeviceConnectionError,
+    DeviceUnsupportedError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.util.network import is_host_valid
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(CONF_USERNAME, default=""): str,
+        vol.Optional(CONF_PASSWORD, default=""): str,
+    }
+)
+
+
+async def validate_input(
+    hass: HomeAssistant, host: str, data: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate the user input."""
+
+    if not is_host_valid(host):
+        raise InvalidHost
+
+    device = await Py2NDevice.create(
+        async_get_clientsession(hass),
+        Py2NConnectionData(
+            host,
+            data.get(CONF_USERNAME),
+            data.get(CONF_PASSWORD),
+        ),
+    )
+
+    return {"title": device.data.name}
```

Comment:
Returning a dict with a single item? Why not just return the title itself directly in that case?

Reply:
That's nonsense, yes. Did some overhauling, now the dict is being properly used!